src/RouteCollector.php

```diff
@@ -49,9 +49,15 @@ class RouteCollector
      */
     private $routes = [];
 
-    public function __construct(RouterInterface $router)
+    /**
+     * @var DuplicateRouteDetector
+     */
+    private $duplicateRouteDetector;
+
+    public function __construct(RouterInterface $router, ?DuplicateRouteDetector $duplicateRouteDetector = null)
     {
         $this->router = $router;
+        $this->duplicateRouteDetector = $duplicateRouteDetector ?? new DuplicateRouteDetector();
```

Comment:
Since `DuplicateRouteDetector` is marked `final`, and has no dependencies, couldn't we omit the constructor argument for it, and just leave this line?

```suggestion
    public function __construct(RouterInterface $router)
    {
        $this->router = $router;
        $this->duplicateRouteDetector = new DuplicateRouteDetector();
```